source/administrator-guide/managing-users-and-groups/authentication-mode.rst

```diff
@@ -510,6 +511,291 @@ You can configure your environment by updating the previous file or by defining
     cas.login.url=${cas.baseURL}/login
     cas.logout.url=${cas.baseURL}/logout?url=${geonetwork.https.url}/
 
+
+.. _authentication-openid:
+
+Configuring OAUTH2 OpenID Connect
+---------------------------------
+
+`OAUTH2 OpenID Connect <https://openid.net/connect/>`_ is an authentication and authorization system based on OAUTH2.  Geonetwork's OpenID Connect plugin has been tested with `Keycloak <https://keycloak.org>`_ and `Azure AD <https://azure.microsoft.com/en-ca/services/active-directory/>`_, but should work with any provider.
+
+Basic Setup Steps;
+
+#. Configure your IDP Server (i.e. Keycloak or Azure AD)
+
+    #. Ensure that the ID Token provides role/group information
+    #. Authorize your Geonetwork URLs for redirects (i.e. `http://localhost:8080/geonetwork/login/oauth2/code/geonetwork-oicd`) 
+    #. Record the Client ID 
+    #. Record the Client Secret
+    #. Get the Server's JSON metadata document
+
+#. Configure Geonetwork via environment variables
+    
+        #. `GEONETWORK_SECURITY_TYPE=openidconnect`
+        #. `OPENIDCONNECT_CLIENTSECRET=...`   (from your IDP server)
+        #. `OPENIDCONNECT_CLIENTID=...`  (from your IDP server)
+        #. `OPENIDCONNECT_SERVERMETADATA_JSON_TEXT='...'` (the text of your Server's JSON metadata document)
+        #. `OPENIDCONNECT_IDTOKENROLELOCATION=...` (location of the user's roles in the ID Token)
+
+
+
+Geonetwork's Open ID Connect plugin has a lot of configuration options - please see the `config-security-openidconnect.xml` and `config-security-openidconnect-overrides.properties` files.
```

Comment:
Better to use to format the file names `` instead of `.

Reference them inside the `WEB-INF/config-security` folder as in other places in the document

Reply:
changed